api/cas-server-core-api-events/src/main/java/org/apereo/cas/support/events/ticket/AbstractCasTicketGrantingTicketEvent.java

```diff
@@ -0,0 +1,74 @@
+package org.apereo.cas.support.events.ticket;
+
+import lombok.Getter;
+import lombok.ToString;
+import org.apereo.cas.support.events.AbstractCasEvent;
+import org.apereo.cas.ticket.TicketGrantingTicket;
+
+import java.time.ZonedDateTime;
+
+/**
+ * Abstract subclass of {@link AbstractCasEvent} encapsulating TGT and exposing key pieces attached to it. Useful for
+ * all TGT-based events.
+ *
+ * @author Dmitriy Kopylenko
+ * @since 6.1.0
+ */
+@ToString(callSuper = true)
+@Getter
+public abstract class AbstractCasTicketGrantingTicketEvent extends AbstractCasEvent {
+
+    public static final long serialVersionUID = 5815205609847140811L;
+
+    private final TicketGrantingTicket ticketGrantingTicket;
+
+    public AbstractCasTicketGrantingTicketEvent(final Object source, final TicketGrantingTicket ticketGrantingTicket) {
```

Comment:
Just thinking out loud, it would be super nice if Lombok could somehow provide a `@InheritedConstructors` annotation :)

Reply:
Is there such a thing?

Reply:
Unfortunately, no. But it's a popular requested item: https://github.com/rzwitserloot/lombok/issues/375